### DISCUSSION TITLE
how to handle python dictionaries in pyccel

### DISCUSSION THREAD

**Discussion by madou-sow** [pyccel/pyccel#1102] [Q&A] [answered]
**
Hello I would like to know how to handle python dictionaries in pyccel
**

```
# test1.py

dict = {"nom": "Gayerie", "prenom": "David"}
vue_dict = dict.keys();
print(vue_dict)
print('\n')
# Affiche dict_keys(['nom', 'prenom'])
vue_dict = dict.items();
print(vue_dict)
print('\n')
# Affiche dict_items([('nom', 'Gayerie'), ('prenom', 'David')])

```

**Comment by EmilyBourne** [comment]
Please see #1053

**Comment by EmilyBourne** [comment]
Currently dictionaries are not implemented in pyccel. We would like to add them, but none of our development team has experience using hashes in Fortran so we are not sure which method is the best or what are the advantages/disadvantages of each method.

We also have a few problems with strings that will need to be tackled before we fix this (strings have not been a priority for us as we are focused on HPC applications, where strings are usually only used in a very basic way (eg for output)). This is demonstrated by our [limited string tests](https://github.com/pyccel/pyccel/blob/master/tests/epyccel/modules/strings.py).

For your example if the keys are always the same it is theoretically possible to handle some of this using classes, e.g:
```python
#$ header class Date(public)
#$ header method __init__(Date, int, int, int)

class Date:
    def __init__(self, jour, mois, annee):
        self.jour = jour
        self.mois = mois
        self.annee = annee

if __name__ == '__main__':
    aujourdhui = Date(11,1,2022)
    print(aujourdhui.jour, aujourdhui.mois, aujourdhui.annee)
```

However the problems with strings mentioned above cause problems when trying to save strings in classes, eg.:
```
#$ header class Name(public)
#$ header method __init__(Name, str, str)

class Name:
    def __init__(self, nom, prenom):
        self.nom = nom
        self.prenom = prenom

if __name__ == '__main__':
    personne_test = Name("Gayerie","David")
    print(personne_test.nom, personne_test.prenom)
```

**Comment by madou-sow** [comment]
Thank you very much

**Comment by EmilyBourne** [comment, marked as answer]
Python classes can now be translated using Pyccel. String keys are still a limitation when translating to Fortran but the C translation works
